Frontend/ui-common/src/Engineering/Helpers/Utils.purs

```diff
@@ -333,3 +333,24 @@ fetchLanguage currLang = case currLang of
                   "TA_IN" -> "ta"
                   _       -> "en"
                   
+getReferralCode :: String -> Maybe String
+getReferralCode referralData =
+  let tuples = parseKeyValues referralData "&" "="
+  in findValueFromTuples tuples "utm_campaign"
+
+parseKeyValues :: String -> String -> String -> Array (Tuple String String)
+parseKeyValues string queryPattern keyValuePattern = DA.catMaybes $ map (\query -> parseKeyValueWithPattern query keyValuePattern) (split (Pattern queryPattern) string)
+
+parseKeyValueWithPattern :: String -> String -> Maybe (Tuple String String)
```

Comment:
`parseKeyValueWithPattern :: String -> String -> Maybe (Tuple String String)
parseKeyValueWithPattern string pattern = 
  case split (Pattern pattern) string of
    [key, value] -> Just (Tuple key value)
    _ -> Nothing`
 
 Try this simplified version